tests/nested/manual/uc20-fde-hooks/task.yaml

```diff
@@ -0,0 +1,51 @@
+summary: Check that the fde-setup hooks work
+
+# this is a UC20 specific test
+systems: [ubuntu-20.04-64]
+
+environment:
+    NESTED_IMAGE_ID: core20-fde-setup
+    NESTED_ENABLE_TPM: false
+    NESTED_ENABLE_SECURE_BOOT: false
+    NESTED_BUILD_SNAPD_FROM_CURRENT: true
+    NESTED_ENABLE_OVMF: true
+    GO111MODULE: off
+
+prepare: |
+  echo "Build a kernel snap with the fde-setup hook"
+  # shellcheck source=tests/lib/prepare.sh
+  . "$TESTSLIB/prepare.sh"
+  # shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+  mkdir ./extra-snaps
+  # build fde-reveal-key hook into the "extra-initrd" dir so that the
+  # nested_create_core_vm picks this up
+  mkdir -p ./extra-initrd/usr/bin/
+  go build -o ./extra-initrd/usr/bin/fde-reveal-key "$TESTSLIB"/fde-setup-hook/fde-setup.go
+
+  # FIXME: hack systemd-run to just run "fde-reveal-key" to workaround
+  # https://github.com/snapcore/core-initrd/issues/13
+  cat > ./extra-initrd/usr/bin/systemd-run <<'EOF'
+  #!/bin/sh
+  exec fde-reveal-key
+  EOF
+  chmod +x ./extra-initrd/usr/bin/systemd-run
+  
+  # create fde-setup hook inside the kernel
+  mkdir -p ./extra-kernel-snap/meta/hooks
+  go build -o ./extra-kernel-snap/meta/hooks/fde-setup "$TESTSLIB"/fde-setup-hook/fde-setup.go
+  nested_create_core_vm
+  # start it
+  nested_start_core_vm
+
+execute: |
+  # shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+  # Wait for snapd to be seeded
+  nested_exec sudo snap wait system seed.loaded
+  # check that we have an encrypted system
+  nested_exec "find /dev/mapper" | MATCH ubuntu-data-[0-9a-f-]+
+  nested_exec "test -e /var/lib/snapd/device/fde/recovery.key"
```

Comment:
maybe it's too much work (and certainly fine to do as a followup), but should we do a simple verification that the key from `snap recovery --show-keys` and this recovery.key match? it shouldn't be _that_ much work since the key is just sealed using xor with `0x13`, but maybe there's something else that complicates this

Reply:
It's a great suggestion! I'll explore it but it might become a followup